sdk/python/feast/infra/online_stores/dynamodb.py

```diff
@@ -309,24 +309,40 @@ def to_tbl_resp(raw_client_response):
                 "values": deserialize(raw_client_response["values"]),
             }
 
+        batches = []
+        entity_id_batches = []
+        while True:
+            batch = list(itertools.islice(entity_ids_iter, batch_size))
+            if not batch:
+                break
+            entity_id_batch = self._to_client_batch_get_payload(
+                online_config, table_name, batch
+            )
+            batches.append(batch)
+            entity_id_batches.append(entity_id_batch)
+
         async with self._get_aiodynamodb_client(online_config.region) as client:
-            while True:
-                batch = list(itertools.islice(entity_ids_iter, batch_size))
-
-                # No more items to insert
-                if len(batch) == 0:
-                    break
-                batch_entity_ids = self._to_client_batch_get_payload(
-                    online_config, table_name, batch
-                )
-                response = await client.batch_get_item(
-                    RequestItems=batch_entity_ids,
-                )
-                batch_result = self._process_batch_get_response(
-                    table_name, response, entity_ids, batch, to_tbl_response=to_tbl_resp
-                )
-                result.extend(batch_result)
-        return result
+            response_batches = await asyncio.gather(
+                *[
+                    client.batch_get_item(
+                        RequestItems=entity_id_batch,
+                    )
+                    for entity_id_batch in entity_id_batches
+                ]
+            )
+
+        result_batches = []
+        for batch, response in zip(batches, response_batches):
+            result_batch = self._process_batch_get_response(
+                table_name,
+                response,
+                entity_ids,
+                batch,
+                to_tbl_response=to_tbl_resp,
+            )
+            result_batches.append(result_batch)
```

Comment:
format the responses to the final format. we iterate through the list three times in stead of one, but make up for it in asyncing the batches